### PR TITLE
feat: added complete test summary report

### PR DESCRIPTION
**Pull request by officialasishkumar**
## Related Issue
  Added complete test summary report

Closes: #1190 

#### Describe the changes you've made
Provided a complete test summary report. Please take a look at the attached screenshot for more clarity.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Please let us know if any test cases are added

Please describe the tests(if any). Provide instructions how its affecting the coverage.

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
NA

A clear and concise description of it.

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

## Screenshots (if any)
image

### CODE REVIEW

**Comment by officialasishkumar** [comment]
@Sonichigo PTAL

**Comment by Sonichigo** [comment]
LGTM! @Sarthak160 can you check once as well

**Comment by slayerjain** [comment]
@shivamsouravjha @Sarthak160 please review this

**Comment by officialasishkumar** [comment]
@shivamsouravjha @Sonichigo  What all information will be on that file test report file?

**Review by shivamsouravjha — APPROVED** [review]
LGTM!

File: pkg/service/test/test.go [hunk 1]
```diff
@@ -260,6 +266,9 @@ func (t *tester) Test(path string, testReportPath string, appCmd string, options
 			break
 		}
 	}
+
+	pp.Printf("\n <=========================================> \n  COMPLETE TESTRUN SUMMARY. \n\tTotal tests: %s\n"+"\tTotal test passed: %s\n"+"\tTotal test failed: %s\n <=========================================> \n\n", totalTests, totalTestPassed, totalTestFailed)
+
 	t.logger.Info("test run completed", zap.Bool("passed overall", result))
 	// log the overall code coverage for the test run of go binaries
```
Comment (shivamsouravjha) [inline pkg/service/test/test.go:273]: please display total number of test suites run and their names  as well

**Review by shivamsouravjha — CHANGES_REQUESTED** [review]

**Comment by officialasishkumar** [comment]
Please take a look @shivamsouravjha

File: pkg/service/test/test.go [hunk 2]
```diff
@@ -55,6 +55,20 @@ type TestOptions struct {
 	PassthroughHosts   []models.Filters
 }
 
+type Tuple struct {
+    First int
```
Comment (shivamsouravjha) [inline pkg/service/test/test.go:59]: tuple seems to be a generic name we want something for out context. Also rename first second to something with Keploy's. context like totaltest passed failed and report structure.

File: pkg/service/test/test.go [hunk 3]
```diff
@@ -257,6 +271,16 @@ func (t *tester) Test(path string, testReportPath string, appCmd string, options
...
 		}
 	}
+
+	pp.Printf("\n <=========================================> \n  COMPLETE TESTRUN SUMMARY. \n\tTotal tests: %s\n"+"\tTotal test passed: %s\n"+"\tTotal test failed: %s\n", totalTests, totalTestPassed, totalTestFailed)
+
+	pp.Printf("\n\tTest Suite Name\t\tTotal Test\tPassed\t\tFailed\t\n")
+	for testSuiteName, verdict := range completeTestReport {
+		pp.Printf("\n\t%s\t\t%s\t\t%s\t\t%s", testSuiteName, verdict.Third, verdict.First, verdict.Second)
+	}
+
```
Comment (shivamsouravjha) [inline pkg/service/test/test.go:281]: print the result of test suites in sorted order

**Review by shivamsouravjha — CHANGES_REQUESTED** [review]

File: pkg/service/test/test.go (hunk 3 above)
Comment (officialasishkumar) [reply pkg/service/test/test.go]: done

File: pkg/service/test/test.go (hunk 2 above)
Comment (officialasishkumar) [reply pkg/service/test/test.go]: done

File: pkg/service/test/test.go [hunk 4]
```diff
@@ -55,6 +55,20 @@ type TestOptions struct {
 	PassthroughHosts   []models.Filters
 }
 
+type TestReportVerdict struct {
+    	total int
+    	passed int
+    	failed int
+}
+
```
Comment (shivamsouravjha) [inline pkg/service/test/test.go:63]: since we've util file this struct can be moved there. Rest LGTM

Comment (officialasishkumar) [reply pkg/service/test/test.go]: done

**Comment by PranshuSrivastava** [comment]
@shivamsouravjha PTAL

**Review by shivamsouravjha — APPROVED** [review]
LGTM

**Comment by shivamsouravjha** [comment]
reverting the merge as unlike pipeline the report is shown irregularly among logs.

**Pull request merged**